server/blob/blob-memory/src/main/java/org/apache/james/blob/memory/MemoryBlobStore.java

```diff
@@ -63,7 +66,14 @@ public MemoryBlobStore(BlobId.Factory factory) {
 
     @Override
     public CompletableFuture<byte[]> readBytes(BlobId blobId) {
-        return CompletableFuture.completedFuture(retrieveStoredValue(blobId));
+        try {
+            return CompletableFuture.completedFuture(retrieveStoredValue(blobId));
+        } catch (ObjectStoreException e) {
+            Supplier<byte[]> throwing = () -> {
+                throw e;
+            };
+            return CompletableFuture.supplyAsync(throwing);
```

Comment:
you cannot use `completeExceptionally(Throwable ex)` instead?

Reply:
```
  CompletableFuture<ClusterConnectionReport> failedFuture = new CompletableFuture<>();
  failedFuture.completeExceptionally(e);
  return failedFuture;
```
 is still shorted, no?

Reply:
Ok, why not. Would you agree if I do that in another PR because I have a chain of dependencies between my PRs that make waiting for this one painful